unit_tests/common/mock_stream.cpp

```diff
@@ -23,7 +25,7 @@ int mock_stream::write (const size_t byte_count, cubstream::stream::write_func_t
 {
   int err;
 
-  err = write_action (last_position, write_buffer, byte_count);
+  err = write_action (last_position, write_buffer + last_position, byte_count);
```

Comment:
I think the old code was correct.

Reply:
well, the old code would overwrite the same last written part of the buffer and implicitly would mess up the postcondition of the test, which checks that the sum is n(n-1)/2.

Reply:
I accept that this is the logic of test.